### PR TITLE
move create and delete obc as part of create_resource and delete_resource…

### LINKED ISSUE #2894: Delete OBC after deployment verification in case of External cluster: SanityExternalCluste() class
E           ocs_ci.ocs.exceptions.CommandFailed: Error during execution of command: oc create -f /tmp/obc_rgw_data443fpwwb.
E           Error is Error from server (AlreadyExists): error when creating "/tmp/obc_rgw_data443fpwwb": objectbucketclaims.objectbucket.io "ceph-delete-bucket" already exists

### PR DESCRIPTION
**Pull request by shylesh**
Moving the create and delete functions for OBC in case of external cluster

Fixes #2894
Signed-off-by: Shylesh Kumar Mohan <shmohan@redhat.com>

### CODE REVIEW

**Review by petr-balogh — DISMISSED** [review]

File: tests/sanity_helpers.py [hunk 1]
```diff
@@ -148,9 +148,27 @@ def __init__(self):
         self.pvc_objs = list()
         self.pod_objs = list()
         self.ceph_cluster = CephClusterExternal()
+
+    def create_resources(self, pvc_factory, pod_factory):
```
Comment (ebenahar) [inline tests/sanity_helpers.py:152]: can we add it also to tests.sanity_helpers.Sanity.create_resources?

Comment (shylesh) [reply tests/sanity_helpers.py:152]: @ebenahar , I didn't get it completely. You meant to move obc deletion to parent class ?

Comment (ebenahar) [reply tests/sanity_helpers.py:152]: I meant that the new functionalities added here would be relevant for internal mode cluster as well

**Comment by shylesh** [comment]
@sidhant-agrawal , https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/12353/consoleFull , I could see successful deletion of obcs. I have to address @ebenahar comment, anything specific need to be handled for converged mode.

I remember @sidhant-agrawal told me that for aws platform condition might be different, it would be great if you can mention the platforms for which its applicable.

**Comment by sidhant-agrawal** [comment]
@shylesh AFAIK, `ceph-rgw` storageclass will be available only in on_prem platforms, i.e., vSphere and Baremetal.

**Comment by shylesh** [comment]
rebased on latest master

File: tests/sanity_helpers.py [hunk 2]
```diff
@@ -66,6 +66,65 @@ def create_resources(self, pvc_factory, pod_factory, run_io=True):
                 pod.run_io('fs', '1G', runtime=30)
             for pod in self.pod_objs:
                 get_fio_rw_iops(pod)
+        self.create_obc()
+
+    def create_obc(self):
+        """
+        OBC creation for RGW and Nooba
+        only applicable for external cluster
```
Comment (ebenahar) [inline tests/sanity_helpers.py:74]: I think this is not relevant now

Comment (shylesh) [reply tests/sanity_helpers.py]: done.

**Comment by shylesh** [comment]
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/12702

**Comment by shylesh** [comment]
@ebenahar @sidhant-agrawal , I ran this on vmware deployment with external mode and it worked. Any other platform we need to check ?

**Comment by shylesh** [comment]
Regression check on AWSIPI : https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-destroy-ocs-cluster/10034/ . Deployment sucessful

**Comment by sidhant-agrawal** [comment]
> @ebenahar @sidhant-agrawal , I ran this on vmware deployment with external mode and it worked. Any other platform we need to check ?

Check on AWS IPI should be enough

**Review by sidhant-agrawal — APPROVED** [review]

**Review by clacroix12 — APPROVED** [review]

**Pull request merged**